core/channel.go

```diff
@@ -74,6 +80,54 @@ func CreateChannel(pathName string, src, dst *ProvableChain, to time.Duration) e
 	return nil
 }
 
+func checkChannelCreateReady(src, dst *ProvableChain, logger *log.RelayLogger) (bool, error) {
+	srcID := src.Chain.Path().ChannelID
+	dstID := dst.Chain.Path().ChannelID
+
+	if srcID == "" && dstID == "" {
+		return true, nil
+	}
+
+	getState := func(pc *ProvableChain) (chantypes.State, error) {
+		if pc.Chain.Path().ChannelID == "" {
+			return chantypes.UNINITIALIZED, nil
+		}
+
+		latestHeight, err := pc.LatestHeight()
+		if err != nil {
+			return chantypes.UNINITIALIZED, err
+		}
+		res, err2 := pc.QueryChannel(NewQueryContext(context.TODO(), latestHeight))
+		if err2 != nil {
+			return chantypes.UNINITIALIZED, err2
+		}
+		return res.Channel.State, nil
+	}
+
+	srcState, srcErr := getState(src)
+	if srcErr != nil {
+		return false, srcErr
+	}
+
+	dstState, dstErr := getState(src)
+	if dstErr != nil {
+		return false, dstErr
+	}
+
+	if srcID != "" && srcState == chantypes.UNINITIALIZED {
+		return false, fmt.Errorf("src channel id is given but that channel does not exist: %s", srcID);
+	}
+	if dstID != "" && dstState == chantypes.UNINITIALIZED {
+		return false, fmt.Errorf("dst channel id is given but that channel does not exist: %s", dstID);
+	}
+
+	if srcState == chantypes.OPEN && dstState == chantypes.OPEN {
+		logger.Warn(fmt.Sprintf("channels are already created: src=%s, dst=%s", srcID, dstID))
```

Comment:
Thank you for using the logger!
However please use structured logging style like shown below.
https://github.com/hyperledger-labs/yui-relayer/blob/v0.5.1/core/naive-strategy.go#L109-L116

Reply:
done be structured.